extractor/src/main/scala/org/jetbrains/sbt/extractors/ProjectExtractor.scala

```diff
@@ -319,6 +344,18 @@ object ProjectExtractor extends SbtStateOps with TaskOps {
       val javaHome = Keys.javaHome.in(projectRef, Compile).value
       val compileOrder = Keys.compileOrder.in(projectRef, Compile).value
 
+      val sourceConfigurations = StructureKeys.sourceConfigurations.value
+      val testConfigurations = StructureKeys.testConfigurations.value
+
+      // note: because we are extracting ConfigurationData with all sourceConfigurations and testConfigurations we also have to take sourceDirectories
```

Comment:
Hm, without knowing all the details this explanation is not much helpful.
As I mentioned in another comment, it would be nice to mention the difference  of `mainSourceDirectories` and `ConfigurationData.sources` in the ScalaDoc with some examples

Reply:
I added ScalaDoc as you suggested (with examples) and removed this note. It's not that important and quite obvious.